File: leetcode2/0083. Remove Duplicates from Sorted List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        node=ListNode(0)
        node.next=head
        p=node
        while p and p.next and p.next.next:
            if p.next.val==p.next.next.val:
                p.next=p.next.next
            else:
                p=p.next
        return node.next

File: archives/C#/0083. Remove Duplicates from Sorted List.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode DeleteDuplicates(ListNode head)
        {
            ListNode node = new ListNode(0);
            node.next = head;
            ListNode p = node;
            while (p.next!=null && p.next.next != null)
            {
                if (p.next.val == p.next.next.val)
                {
                    p.next = p.next.next;
                }
                else
                {
                    p = p.next;
                }
            }
            return node.next;

        }
}
